test/api/controllers/validators/ValidatorSpec.scala

```diff
@@ -0,0 +1,172 @@
+  }
+
+  "validator.parseAndValidateRequest()" should {
+
+    "return the parsed domain object with no errors" when {
+      "all data is correct" in new Test {
+        val result: Either[ErrorWrapper, TestParsedRequest] = validator.parseAndValidateRequest(validRaw)
+        result shouldBe Right(parsed)
+      }
+    }
+
+    "return a single error" when {
+      "just one validation error is found" in new Test {
+        override lazy val parserValidation: ParserValidationCaller[TestRawData, TestParsedRequest] =
+          _ => Left(List(NinoFormatError))
+
+        val result: Either[ErrorWrapper, TestParsedRequest] = validator.parseAndValidateRequest(validRaw)
+        result shouldBe Left(ErrorWrapper(correlationId, NinoFormatError, None))
+      }
+    }
+
+    "return multiple errors" when {
+      "both the param and body are invalid" in new Test {
+
+        override lazy val preParseValidations: PreParseValidationCallers[TestRawData] =
+          List(
+            _ => List(NinoFormatError),
+            _ => List(RuleIncorrectOrEmptyBodyError)
+          )
+
+        val result: Either[ErrorWrapper, TestParsedRequest] = validator.parseAndValidateRequest(validRaw)
+        result shouldBe Left(ErrorWrapper(correlationId, BadRequestError, Some(List(NinoFormatError, RuleIncorrectOrEmptyBodyError))))
+      }
+    }
+  }
+
+  "validator.parseAndValidate()" should {
+
+    "return the parsed domain object with no errors" when {
+      "all data is correct" in new Test {
+        val levelOneValidationOne = new MockFunctionObject
+        val levelOneValidationTwo = new MockFunctionObject
+
+        def levelOneValidations: PostParseValidationCaller[TestParsedRequest] =
+          _ =>
+            levelOneValidationOne.noErrors() ++
+              levelOneValidationTwo.noErrors()
+
+        override lazy val postParseValidations = List(levelOneValidations)
+
+        val result: Either[Seq[MtdError], TestParsedRequest] = validator.parseAndValidate(validRaw)
+        result shouldBe Right(parsed)
+        levelOneValidationOne.called shouldBe 1
+        levelOneValidationTwo.called shouldBe 1
+      }
+    }
+
+    "return a list of validation errors on level one" when {
+      "there are failed validations" in new Test {
+
+        val levelOneValidationOne = new MockFunctionObject
+        val levelOneValidationTwo = new MockFunctionObject
+
+        override lazy val postParseValidations = List(levelOneValidations)
+
+        val mockError: MtdError = MtdError("MOCK", "SOME ERROR", CONFLICT)
+
+        def levelOneValidations: PostParseValidationCaller[TestParsedRequest] =
+          _ =>
+            levelOneValidationOne.noErrors() ++
+              levelOneValidationTwo.error(mockError)
+
+        val result: Either[Seq[MtdError], TestParsedRequest] = validator.parseAndValidate(validRaw)
+        result shouldBe Left(List(mockError))
+
+        levelOneValidationOne.called shouldBe 1
+        levelOneValidationTwo.called shouldBe 1
```

Comment:
Can we add tests here to check that levelTwo validations are not being called if there are errors returned for levelOne validations please?